### PR TITLE
DB Progress

### PR DESCRIPTION
**Pull request by stingray127**
Cards show up, working on Phase 4

### CODE REVIEW

File: services/board.js [hunk 1]
```diff
@@ -115,15 +119,24 @@ module.exports = function () {
                 if (this.phase !== this.Phases.four) {
                     return false;
                 }
-                this.display.blackCard = jsonHandler.createBlackCard();
+                var display = this.display;
+                var self = this;
```
Comment (gewenyu99) [inline services/board.js]: You don't need to do this, just pass in a reference to this.display to your function to preserve context.

File: services/board.js [hunk 2]
```diff
@@ -20,7 +20,7 @@ module.exports = function () {
             phase: 0,
             players: {},
             display: {
-                "blackCard": null, //This should be a black card object
+                "blackCard": '', //This should be a black card object
```
Comment (gewenyu99) [inline services/board.js]: why

File: services/dbService.js [hunk 3]
```diff
@@ -2,12 +2,14 @@ var instance;
 module.exports = function () {
     if(!instance){
         instance = {
+            connection: 0,
```
Comment (gewenyu99) [inline services/dbService.js]: : null

File: services/dbService.js [hunk 4]
```diff
@@ -18,11 +20,19 @@ module.exports = function () {
...
-            getWhiteCards: function () { // gets rando white card, or many cards
+            getWhiteCard: function (id, callback) {
+                connection.query('SELECT * FROM white_card', function(err, results, fields){
+                    if(err) throw err;
 
+                    callback(results[0]);
+                });
             },
-            getBlackCard: function () { //gets rand black card
+            getBlackCard: function (id, callback) { //gets rand black card
```
Comment (gewenyu99) [inline services/dbService.js]: Sorry for my ignorance, but is id used here?

File: services/board.js (hunk 1 above)
Comment (stingray127) [reply services/board.js]: So add another parameter? I wasn't sure if the scope would be correct.

File: services/board.js (hunk 2 above)
Comment (stingray127) [reply services/board.js]: fine

File: services/dbService.js (hunk 3 above)
Comment (stingray127) [reply services/dbService.js]: o k a y

File: services/dbService.js (hunk 4 above)
Comment (stingray127) [reply services/dbService.js]: Yeah id is a random number

File: services/board.js [hunk 5]
```diff
@@ -115,15 +119,24 @@ module.exports = function () {
...
                 var key;
                 var keys = Object.keys(this.players);
                 //console.log(keys);
                 for (key in keys) {
                     //console.log(key);
                     if (key !== this.display.currentJudge) {
-                        this.players[keys[key]].data.hand.push(jsonHandler.createWhiteCard(keys[key]));
+                        console.log(key);
+                        jsonHandler.createWhiteCard(key, function(card){
+                            self.players[keys].data.hand.push();
```
[the remark this answers is not in the document]
Comment (stingray127) [reply services/board.js]: Yeah phase 4 is a shitshow

**Pull request closed without merging**